### DISCUSSION TITLE
Associations

### DISCUSSION THREAD

**Discussion by fabnavigator** [zwave-js/node-red-contrib-zwave-js#253] [Q&A]
Is there a way to know which devices can be controlled by an association. I've tried three devices, and one works and two don't.

**Comment by marcus-j-davies** [comment]
The association UI will show what the device supports (or tells us about at least) some devices only have 1 - the lifeline, meaning it can only send to the controller, if it has more than 1 then the other groups can be used to tell other devices about an event without the controller or NR needing to be online as its device to device.

Example:
A motion sensor might have a group called "security" - that say, can send a binary on/off message to target nodes (I.e switching on a siren let's say) - some devices allow you to set the CC via configuration cc

The CCs that are used for the notification will be documented in the devices manual (there is no agreement in spec what is actually sent from an association)

Of course - the devices receiving them (I.e those mentioned in your associated list) must support the CC. A device receiving the message it's self, does not need to support association, to be able to be a target node, for a device that does.

All devices will usually have a lifeline group - i.e addressed to the controller.

In other words - a receiving device just needs to support the CC that is used, and the sending device support associations.

**Reply by fabnavigator** [reply]
Interesting, I can control a binary switch from another binary switch, but my door sensor doesn't have the binary switch class so it can't control a binary switch. Correct?

**Reply by marcus-j-davies** [reply]
> Interesting, I can control a binary switch from another binary switch

Providing it has association groups to do so, and the other target nodes (receiving the event) supports binary cc (on assumption the association message is using binary)

> but my door sensor doesn't have the binary switch class so it can't control a binary switch. Correct?


Correct, if the door sensor does not use Binary in association messages, then having it associated to a device that expects binary will be pointless - the receiving device will just ignore it.


Again, some devices allow setting the CC in its config params - but its per device/manufacture on what they allow the user to configure

**Reply by marcus-j-davies** [reply]
You can test the association groups - just set the controller to be a target node of a group 😄 
Your NR flow will receive 2 events of course - 1 from the lifeline,  1 from the other association group

**Reply by fabnavigator** [reply]
Thank you. I have a better understanding now. I would think that someone would want a door sensor to be able to turn on a lignt (binary switch). Do you know of any that support that command class?

**Reply by marcus-j-davies** [reply]
I dont.
I use the Aeotec Door Sensors - but they only send BASIC CC values for associations sadly.
there are a number Door Sensors that obviously will do.

Search the term Door Sensor over at https://devices.zwave-js.io 

I rarely use D2D - as I like to control all aspects.

DO Note: The switches them self might actually support BASIC
